test/helpers/query.cast$expr.test.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const { Schema } = require('../common').mongoose;
+const assert = require('assert');
+const cast$expr = require('../../lib/helpers/query/cast$expr');
+
+describe('castexpr', function() {
+  it('casts comparisons', function() {
+    const testSchema = new Schema({ date: Date, spent: Number, budget: Number });
+
+    let res = cast$expr({ $eq: ['$date', '2021-06-01'] }, testSchema);
+    assert.deepEqual(res, { $eq: ['$date', new Date('2021-06-01')] });
+
+    res = cast$expr({ $eq: [{ $year: '$date' }, 2021] }, testSchema);
+    assert.deepStrictEqual(res, { $eq: [{ $year: '$date' }, 2021] });
+
+    res = cast$expr({ $eq: [{ $year: '$date' }, '2021'] }, testSchema);
+    assert.deepStrictEqual(res, { $eq: [{ $year: '$date' }, 2021] });
+
+    res = cast$expr({ $eq: [{ $year: '$date' }, { $literal: '2021' }] }, testSchema);
+    assert.deepStrictEqual(res, { $eq: [{ $year: '$date' }, { $literal: 2021 }] });
+
+    res = cast$expr({ $eq: [{ $year: '$date' }, { $literal: '2021' }] }, testSchema);
+    assert.deepStrictEqual(res, { $eq: [{ $year: '$date' }, { $literal: 2021 }] });
+
+    res = cast$expr({ $gt: ['$spent', '$budget'] }, testSchema);
+    assert.deepStrictEqual(res, { $gt: ['$spent', '$budget'] });
+  });
+
+  it('casts conditions', function() {
```

Comment:
Would be nice if we could support `$cond: [myCondition, trueCase, falseCase]` as well.

Would happily work on that in a next PR or on this one. @vkarpov15

Reply:
Fixed in 29b323c